motionplan/motionPlanner.go

```diff
@@ -381,3 +383,125 @@ IK:
+
+	// construct solverFrame
+	// Note that this requires all frames which move as part of the plan, to have an
+	// entry in the very first plan waypoint
+	sf, err := newSolverFrame(fs, checkFrame.Name(), frame.World, plan[0])
+	if err != nil {
+		return err
+	}
+
+	// construct planager
+	sfPlanner, err := newPlanManager(sf, fs, logger, defaultRandomSeed)
+	if err != nil {
+		return err
+	}
+
+	// convert plan into nodes
+	planNodes, err := sf.planToNodes(plan)
+	if err != nil {
+		return err
+	}
+
+	// This should be done for any plan whose configurations are specified in relative terms rather than absolute ones.
+	// Currently this is only TP-space, so we check if the PTG length is >0.
+	// The solver frame will have had its PTGs filled in the newPlanManager() call, if applicable.
+	relative := len(sf.PTGs()) > 0
+
+	if relative {
+		// get pose of robot along the current trajectory it is executing
+		lastPose, err := sf.Transform(currentInputs)
+		if err != nil {
+			return err
+		}
+
+		// where ought the robot be on the plan
+		pathPosition := spatialmath.PoseBetweenInverse(errorState, currentPosition)
+
+		// absolute pose of the previous node we've passed
+		formerRunningPose := spatialmath.PoseBetweenInverse(lastPose, pathPosition)
+
+		// convert planNode's poses to be in absolute corrdinated
+		if planNodes, err = rectifyTPspacePath(planNodes, sf, formerRunningPose); err != nil {
+			return err
+		}
+	}
+	// adjust planNodes by the errorState
+	planNodes = transformNodes(planNodes, errorState)
+
+	// pre-pend node with current position of robot to planNodes
+	// Note that currentPosition is assumed to have already accounted for the errorState
+	planNodes = append([]node{&basicNode{pose: currentPosition, q: currentInputs}}, planNodes...)
+
+	// create constraints
+	if sfPlanner.planOpts, err = sfPlanner.plannerSetupFromMoveRequest(
+		currentPosition,                    // starting pose
+		planNodes[len(planNodes)-1].Pose(), // goalPose
+		plan[0],                            // starting configuration
+		worldState,
+		nil, // no pb.Constraints
+		nil, // no plannOpts
+	); err != nil {
+		return err
+	}
+
+	// go through plan and check that we can move from plan[i] to plan[i+1]
+	for i := 0; i < len(planNodes)-1; i++ {
+		currentPose := planNodes[i].Pose()
+		nextPose := planNodes[i+1].Pose()
+		startConfiguration := planNodes[i].Q()
+		endConfiguration := planNodes[i+1].Q()
+
+		if relative {
+			startConfiguration = []frame.Input{
+				{Value: endConfiguration[0].Value}, {Value: endConfiguration[1].Value}, {Value: 0},
+			}
+		}
+		segment := &ik.Segment{
+			StartPosition:      currentPose,
+			EndPosition:        nextPose,
+			StartConfiguration: startConfiguration,
+			EndConfiguration:   endConfiguration,
+			Frame:              sf,
+		}
+		interpolatedConfigurations, err := getInterpolations(segment, sfPlanner.planOpts.Resolution)
+		if err != nil {
+			return err
+		}
+		for _, interpConfig := range interpolatedConfigurations {
```

Comment:
I'm a little confused about this.  Shouldn't we be able to operate on this segment directly rather than doing this here?